src/test/lib/statistics/cardinality_estimator_test.cpp

```diff
@@ -913,4 +914,150 @@ TEST_F(CardinalityEstimatorTest, NonQueryNodes) {
+        PredicateCondition::GreaterThan, PredicateCondition::GreaterThanEquals}) {
+    const auto predicate =
+        std::make_shared<BinaryPredicateExpression>(predicate_condition, a_a, lqp_subquery_(subquery));
+    lqp = PredicateNode::make(predicate, node_a);
+
+    EXPECT_FLOAT_EQ(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(node_a));
+  }
+}
+
+TEST_F(CardinalityEstimatorTest, BetweenScanWithUncorrelatedSubquery) {
+  // Case (ii): Predicate column BETWEEN min(<subquery>) and max(<subquery>)
+  // Example query:
+  //     SELECT SUM(ws_ext_sales_price) FROM web_sales
+  //      WHERE ws_sold_date_sk BETWEEN (SELECT MIN(d_date_sk) FROM date_dim WHERE d_year = 2000)
+  //                                AND (SELECT MAX(d_date_sk) FROM date_dim WHERE d_year = 2000);
+  const auto subquery = PredicateNode::make(between_inclusive_(c_y, value_(0), value_(2)), node_c);
+
+  const auto min_c_y = AggregateNode::make(expression_vector(), expression_vector(min_(c_y)), subquery);
+
+  const auto max_c_y = AggregateNode::make(expression_vector(), expression_vector(max_(c_y)), subquery);
+
+  auto lqp = PredicateNode::make(between_inclusive_(a_a, lqp_subquery_(min_c_y), lqp_subquery_(max_c_y)), node_a);
+
+  const auto semi_join_lqp = JoinNode::make(JoinMode::Semi, equals_(a_a, c_y), node_a, subquery);
+
+  EXPECT_FLOAT_EQ(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(semi_join_lqp));
+  EXPECT_LT(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(node_a));
+
+  // Ensure correlated subqueries are not considered and we forward the input estimates.
+  // clang-format off
+  lqp =
+  PredicateNode::make(between_inclusive_(a_a, lqp_subquery_(min_c_y, std::make_pair(ParameterID{0}, a_b)), lqp_subquery_(max_c_y, std::make_pair(ParameterID{0}, a_b))),  // NOLINT(whitespace/line_length)
+    node_a);
+  // clang-format on
+
+  // Ensure we do not consider the wrong aggregate functions and forward the input estimates.
+  lqp = PredicateNode::make(between_inclusive_(a_a, lqp_subquery_(max_c_y), lqp_subquery_(max_c_y)), node_a);
+  EXPECT_FLOAT_EQ(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(node_a));
+
+  lqp = PredicateNode::make(between_inclusive_(a_a, lqp_subquery_(min_c_y), lqp_subquery_(min_c_y)), node_a);
+  EXPECT_FLOAT_EQ(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(node_a));
+
+  // Aggregate functions must be performed on the same column.
+  const auto min_c_x = AggregateNode::make(expression_vector(), expression_vector(min_(c_x)), subquery);
+  lqp = PredicateNode::make(between_inclusive_(a_a, lqp_subquery_(min_c_x), lqp_subquery_(max_c_y)), node_a);
+  EXPECT_FLOAT_EQ(estimator.estimate_cardinality(lqp), estimator.estimate_cardinality(node_a));
+
+  // Aggregate functions must not group the tuples. Otherwise, we do not get the minimum and maximum value for the join
+  // key for the underlying node.
```

Comment:
of the underlying?
The group by restriction assumes that there is more than one group, right?

Reply:
Can you please elaborate?